diff --git a/neutron/notifiers/nova.py b/neutron/notifiers/nova.py
index 48f959e..7354ec0 100644
--- a/neutron/notifiers/nova.py
+++ b/neutron/notifiers/nova.py
@@ -94,7 +94,8 @@ class Notifier(object):
     def _is_compute_port(self, port):
         try:
             if (port['device_id'] and uuidutils.is_uuid_like(port['device_id'])
-                    and port['device_owner'].startswith('compute:')):
+                    and (port['device_owner'].startswith('compute:')
+                    or port['device_owner'].startswith('network:'))):
                 return True
         except (KeyError, AttributeError):
             pass
