diff --git a/neutron/db/models_v2.py b/neutron/db/models_v2.py
index a2f1e03..ba7fcf3 100644
--- a/neutron/db/models_v2.py
+++ b/neutron/db/models_v2.py
@@ -80,7 +80,6 @@ class IPAllocationPool(model_base.BASEV2, HasId):
     last_ip = sa.Column(sa.String(64), nullable=False)
     available_ranges = orm.relationship(IPAvailabilityRange,
                                         backref='ipallocationpool',
-                                        lazy="joined",
                                         cascade='delete')

     def __repr__(self):
